src/lib/greenboot.rs

```diff
@@ -233,11 +263,33 @@ mod test {
         tear_down().context("Test teardown failed").unwrap();
     }
 
+    #[test]
+    fn test_skip_failing_binary() {
+        setup_folder_structure(false)
+            .context("Test setup failed")
+            .unwrap();
+
+        // Only testing binary in this test. Including extra failing script causes failure.
+        let required_path = format!("{}/check/required.d", GREENBOOT_INSTALL_PATHS[1]);
+        let _ = std::fs::remove_file(format!("{}/failing_script.sh", required_path));
+
+        // Skip the failing script in required.d
+        let state = run_diagnostics(vec!["failing_binary".to_string()]);
+        assert!(
+            state.is_ok(),
+            "Should pass when skipping failing required binary"
+        );
+
+        tear_down().context("Test teardown failed").unwrap();
+    }
+
     fn setup_folder_structure(passing: bool) -> Result<()> {
         let required_path = format!("{}/check/required.d", GREENBOOT_INSTALL_PATHS[1]);
         let wanted_path = format!("{}/check/wanted.d", GREENBOOT_INSTALL_PATHS[1]);
         let passing_test_scripts = "testing_assets/passing_script.sh";
         let failing_test_scripts = "testing_assets/failing_script.sh";
+        let passing_test_binary = "testing_assets/passing_binary";
+        let failing_test_binary = "testing_assets/failing_binary";
```

Comment:
@arortiz-rh you will need to add the binaries here too: https://github.com/fedora-iot/greenboot-rs/tree/main/testing_assets